pkg/state/create.go

```diff
@@ -173,13 +173,13 @@ func (c *StateCreator) ParseAndLoad(content []byte, baseDir, file string, envNam
 			return nil, errors.New("nested `base` helmfile is unsupported. please submit a feature request if you need this!")
 		}
 	} else {
-		state, err = c.loadBases(envValues, state, baseDir)
+		state, err = c.loadBases(envValues, overrode, state, baseDir)
```

Comment:
Note: So we now override the env within the nested function call (of loadBases -> loadFile) instead of doing it before calling ParseAndLoad. Gotcha.